### PR TITLE
allow x-example

### PR DESCRIPTION
**Pull request by pouledodue**
the acutal code is searching for 'example' in the parameters. When using the example field like this...
```
  /users/{uid}:
    get:
      summary: Returns a user
      operationId: swagger.api.read_user_by_uid
      produces:
        - application/json
      parameters:
        - name: uid
          in: path
          description: UID of user that needs to be fetched
          required: true
          type: string
#          default: myuuid
          example: myuuid
      responses:
        "200":
          description: One User
          schema:
            $ref: "#/definitions/UserRead"
```
...it fails . The only way to get {uid} populated with 'myuuid' is by using the 'default' property
but it is not intended to be used for setting a default value in such a scenario.

As x-vendor properties are permitted by the spec, let use them:

```
 /users/{uid}:
    get:
      summary: Returns a user
      operationId: swagger.api.read_user_by_uid
      produces:
        - application/json
      parameters:
        - name: uid
          in: path
          description: UID of user that needs to be fetched
          required: true
          type: string
          x-example: myuuid
      responses:
        "200":
          description: One User
          schema:
            $ref: "#/definitions/UserRead"
```

### CODE REVIEW

**Comment by beanqueen** [comment]
i don't fully understand the problem, but i'm ok with the solution anyway. the if / if / elif / .. / else should be refactored to be consistent.

**Comment by cyprieng** [comment]
I don't see why the 'example' field wouldn't work in your example, neither why x-example would improve this. Can you provide the code that fails ?

File: swagger_parser/swagger_parser.py [hunk 1]
```diff
@@ -160,6 +160,8 @@ def get_example_from_prop_spec(self, prop_spec):
         """
         if 'example' in prop_spec.keys() and self.use_example:  # From example
             return prop_spec['example']
+        if 'x-example' in prop_spec.keys() and self.use_example:
```
Comment (cyprieng) [inline swagger_parser/swagger_parser.py]: Can you change `if` to `elif` ?

**Comment by beanqueen** [comment]
we can close this PR, i added the x-example support and cleaned up the if/elif block

**Comment by cyprieng** [comment]
Done in #26

**Pull request closed without merging**